src/components/FilteredSearchBar.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+import React, { PureComponent, Fragment } from "react";
+import { StyleSheet, TextInput, View, TouchableOpacity } from "react-native";
+import throttle from "lodash/throttle";
+
+import Search from "./Search";
+import SearchIcon from "../images/icons/Search";
+import Close from "../images/icons/Close";
+
+import colors from "../colors";
+
+type Props = {
+  renderList: (list: Array<*>) => React$Node,
+  renderEmptySearch: () => React$Node,
+  keys?: Array<string>,
+  list: Array<*>,
+};
+
+type State = {
+  focused: boolean,
+  query: string,
+};
+
+class FilteredSearchBar extends PureComponent<Props, State> {
+  static defaultProps = {
+    keys: ["name"],
+  };
+
+  state = {
+    focused: false,
+    query: "",
+  };
+
+  constructor(props: *) {
+    super(props);
+
+    this.onChange = throttle(this.onChange, 200);
+  }
+
+  // $FlowFixMe
+  input = React.createRef();
+
+  onFocus = () => this.setState({ focused: true });
+  onBlur = () => this.setState({ focused: false });
+
+  onChange = (text: string) => this.setState({ query: text });
+
+  clear = () => {
+    this.input.current.clear();
+    this.onChange("");
```

Comment:
polish: you also want to `this.onChange.cancel()`

Reply:
same here